registry/api/errcode/errors.go

```diff
@@ -97,9 +104,21 @@ func (e Error) Error() string {
 		e.Code.Message())
 }
 
-// Message returned the human-readable error message for this Error
-func (e Error) Message() string {
-	return e.Code.Message()
+// AddDetail will add some Detail info to the Error
+func (e Error) AddDetail(detail interface{}) Error {
```

Comment:
@duglin Creating temporary ones is okay. Supporting inline modification will just lead to issues.